### PR TITLE
feat: add library context identifier

### PR DESCRIPTION
**Pull request by liuyang1520**
### Summary
- feat: add library context identifier

Add this option in the configuration, which can be used in the library wrapper/code generation/dynamic loading cases, for setting the `library_context` information.

### Checklist
* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Python/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: src/amplitude/config.py [hunk 1]
```diff
@@ -71,6 +72,7 @@ def __init__(self, api_key: str = None,
         self.storage_provider: StorageProvider = storage_provider
         self.opt_out: bool = False
         self.plan: Plan = plan
+        self.library_context: str = library_context
```
Comment (bohan-amplitude) [inline src/amplitude/config.py:75]: Can you add the new argument to class docstring too?

File: src/amplitude/event.py [hunk 2]
```diff
@@ -239,6 +240,7 @@ def __init__(self, user_id: Optional[str] = None,
         self.session_id: Optional[int] = None
         self.insert_id: Optional[str] = None
         self.library: Optional[str] = None
+        self.library_context: Optional[str] = None
```
Comment (bohan-amplitude) [inline src/amplitude/event.py:243]: Is the library_context designed to not be part of the constructor parameters?

File: src/amplitude/config.py (hunk 1 above)
Comment (liuyang1520) [reply src/amplitude/config.py:75]: Thanks for reviewing! Will update!

File: src/amplitude/event.py (hunk 2 above)
Comment (liuyang1520) [reply src/amplitude/event.py:243]: Correct me if I am wrong, I think Python SDK is different, it has `EventOptions`, I didn't find similar things in other SDKs. In my understanding, it is a one-time overwrite to provide additional information to the event. If this is the case, I think we probably don't need this in the constructor, as the `library_context` should be always a global information. What is you suggestion on this?

Comment (bohan-amplitude) [reply src/amplitude/event.py:243]: EventOptions are introduced in our new gen SDK e.g. [ts](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-types/src/base-event.ts) [kotlin](https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/events/EventOptions.kt). It's designed for ampli that have all attributes except those controlled by tracking plan. Usually used to provide additional info to events but limit the ability to override ampli controlled part. It's also base class of other event types classes. There are some fields that we used internally like library and plan. Maybe it's ok to not include them in the constructor. I think adding library_context as optional parameter for consistency.

Comment (liuyang1520) [reply src/amplitude/event.py:243]: Got it, thanks for the explanation! Then I will keep it excluded from the constructor like what we have here now.

**Review by bohan-amplitude — APPROVED** [review]

**Review by bohan-amplitude — APPROVED** [review]

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**